source/DSCResources/DSC_CMSystemDiscovery/DSC_CMSystemDiscovery.psm1

```diff
@@ -0,0 +1,598 @@
+
+    .PARAMETER TimeSinceLastLogonDays
+        Specify the number of days for EnableFilteringExpiredLogon.
+
+    .PARAMETER EnableFilteringExpiredPassword
+        Indicates whether Configuration Manager discovers only computers that have updated their computer
+        account password within a specified number of days. Specify the number of days by using the
+        TimeSinceLastPasswordUpdateDays parameter.
+
+    .PARAMETER TimeSinceLastPasswordUpdateDays
+        Specify the number of days for EnableFilteringExpiredPassword.
+
+    .PARAMETER ADContainers
+        Specifies an array of names of Active Directory containers to match to the discovery.
+
+    .PARAMETER ADContainersToInclude
+        Specifies an array of names of Active Directory containers to add to the discovery.
+
+    .PARAMETER ADContainersToExclude
+        Specifies an array of names of Active Directory containers to exclude to the discovery.
+
+    .PARAMETER ScheduleInterval
+        Specifies the time when the scheduled event recurs.
+
+    .PARAMETER ScheduleCount
+        Specifies how often the recur interval is run.
+#>
+function Test-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Boolean])]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteCode,
+
+        [Parameter(Mandatory = $true)]
+        [Boolean]
+        $Enabled,
+
+        [Parameter()]
+        [Boolean]
+        $EnableDeltaDiscovery,
+
+        [Parameter()]
+        [ValidateRange(1,60)]
+        [UInt32]
+        $DeltaDiscoveryMins,
+
+        [Parameter()]
+        [Boolean]
+        $EnableFilteringExpiredLogon,
+
+        [Parameter()]
+        [UInt32]
+        $TimeSinceLastLogonDays,
+
+        [Parameter()]
+        [Boolean]
+        $EnableFilteringExpiredPassword,
+
+        [Parameter()]
+        [UInt32]
+        $TimeSinceLastPasswordUpdateDays,
+
+        [Parameter()]
+        [String[]]
+        $ADContainers,
+
+        [Parameter()]
+        [String[]]
+        $ADContainersToInclude,
+
+        [Parameter()]
+        [String[]]
+        $ADContainersToExclude,
+
+        [Parameter()]
+        [ValidateSet('None','Days','Hours','Minutes')]
+        [String]
+        $ScheduleInterval,
+
+        [Parameter()]
+        [UInt32]
+        $ScheduleCount
+    )
+
+    Import-ConfigMgrPowerShellModule -SiteCode $SiteCode
+    Set-Location -Path "$($SiteCode):\"
+    $state = Get-TargetResource -SiteCode $SiteCode -Enabled $Enabled
+    $result = $true
+
+    if ($Enabled -eq $true)
+    {
+        $testParams = @{
+            CurrentValues = $state
+            DesiredValues = $PSBoundParameters
+            ValuesToCheck = @('Enabled','EnableDeltaDiscovery','DeltaDiscoveryMins','EnableFilteringExpiredLogon',
+                              'TimeSinceLastLogonDays','EnableFilteringExpiredPassword','TimeSinceLastPasswordUpdateDays')
+        }
+
+        $result = Test-DscParameterState @testParams -Verbose
```

Comment:
We will either need to ignore type here or typecast in get. I am testing and got type mismatches for EnableDeltaDiscovery, EnableFilteringExpiredLogon, and EnableFilteringExpiredPassword